test.sh

```
@@ -24,12 +30,92 @@ function prepareOutputDir() {
+
+  # copy test-policy into a /tmp/test-policy within the client container
+  docker cp test-policy ${dap_client_cid}:/tmp
+
+  docker-compose exec client-dap bash -c "conjur policy load root /tmp/test-policy/root.yml"
+}
+
+function initializeDAPCert () {
+  dap_client_cid=$(docker-compose ps -q client-dap)
+    # get the pem file from conjur server
+  CONJUR_ACCOUNT="cucumber"
+  CONJUR_PROXY="https://cuke-master"
+
+#  NOT WORKING
+  echo "remove old pem file"
+  rm -rf /test-cert/*
+
+  echo "fetch pem file from proxy https server"
+  exec_command='echo yes | conjur init -u '${CONJUR_PROXY}' -a '${CONJUR_ACCOUNT}' > tmp.out 2>&1'
+  docker exec ${dap_client_cid} /bin/bash -c "$exec_command"
+
+  echo "print command output"
+  print_command="cat tmp.out"
+  docker exec ${dap_client_cid} ${print_command}
+
+  echo "convert pem to der file and copy it to share memory"
+  convert_command="openssl x509 -outform der -in /root/conjur-cucumber.pem -out /test-cert/conjur-cucumber.der"
+  docker exec ${dap_client_cid} ${convert_command}
+
+  echo "import cert inside DAP test container"
+
+  # get conjur test-dap container id
+  dap_test_cid=$(docker-compose ps -q test-dap)
+
+  JRE_HOME='/usr/lib/jvm/java-8-openjdk-amd64/jre'
+  import_command="keytool -import -alias cucumber -v -trustcacerts -noprompt \
+     -keystore $JRE_HOME/lib/security/cacerts \
+     -file /conjurinc/api-java/test-cert/conjur-cucumber.der -storepass changeit"
+  docker exec ${dap_test_cid} ${import_command}
+}
+
+function runDAPTests() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Running DAP tests"
+  echo '------------------------------------------------------------'
+
+  dap_test_cid=$(docker-compose ps -q test-dap)
+
+  export CONJUR_AUTHN_API_KEY=$(docker-compose exec client-dap conjur user rotate_api_key)
+
+  # Execute tests
+  docker exec \
+    -e CONJUR_AUTHN_API_KEY \
+    -e CONJUR_AUTHN_LOGIN="admin" \
+    ${dap_test_cid} bash -c "mvn test"
```

Is there a need to wrap this with `bash -c`?

---

As a sidenote, when you run `bash -c` you are (like the first line in this file) ignoring any errors that occur in the running of the command. Think about if that's desired here or not.